mrmustard/lab/states/dm.py

```diff
@@ -294,15 +290,13 @@ def expectation(self, operator: CircuitComponent):
 
         leftover_modes = self.wires.modes - operator.wires.modes
         if op_type is OperatorType.KET_LIKE:
-            result = self >> operator.dual
-            if leftover_modes:
-                result >>= TraceOut(leftover_modes)
+            result = self.contract(operator.dual.adjoint, mode=mode).contract(
+                operator.dual, mode="zip"
+            ) >> TraceOut(leftover_modes)
```

Comment:
If `mode='kron'` and both `self` and `operator` have a batch dimension, say `(4,)` and `(7,)`, then the final 'zip' should fail because the caller has batch shape (4,7), and operator.dual has shape (7,). I haven't actually tried, I'm just asking.